ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -579,10 +579,6 @@ class JITModBuilder {
     IntegerType *Int32Ty = Type::getInt32Ty(Context);
     PointerType *Int8PtrTy = Type::getInt8PtrTy(Context);
 
-    // Create struct type for storing a vector and its length.
```

Comment:
I don't understand this commit.

> According to the LLVM docs "opaque pointer mode is automatically disabled for IR and bitcode files that explicitly mention i8* style typed pointers".

This is talking about old-style LLVM pointer types, e.g. `i8*` and new-style ones `ptr`. The latter being opaque.

AFAIK, the only reason we don't use opaque pointers for the JIT, is that many of the intrinsics we use still have an old-style pointer in their signature.

But I don't see any opaque pointers being removed here. What have I missed?

Reply:
Previous we've used `VecPtrTy` for all those `Vec/Len` structs. The opaque pointer type of which is created just underneath your comment ;-) See https://github.com/ykjit/yk/pull/557#diff-b76ed8a29ed650e7c2a44a5809bc05a22cca958d0b9f5532dab74cce8503d112L582. `PointerType::get(Context, 0);` creates an opaque pointer.

My understanding as to why opaque pointer mode is turned off is because we are using `i8*` pointers throughout this code. But I may have misunderstood the LLVM docs. If we can we should probably just replace everything with opaque pointers as LLVM recommends. But that would be in a different PR anyway, if it's possible.

Reply:
I suggest we don't move to opaque pointers just yet, as patchpoints, which we theorise we will need, has this signature:

```
declare void
  @llvm.experimental.patchpoint.void(i64 <id>, i32 <numBytes>,
                                     i8* <target>, i32 <numArgs>, ...)
```

Note the `i8*`. This would stop us using opaques.

Reply:
So really, this commit just needs an updated commit message I think?

Reply:
What do you want to update about the commit message? Does it say something wrong?

Reply:
Ah sorry. So it does indeed remove opaque pointers.